contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py

```diff
@@ -121,22 +152,97 @@ def emit(self, span_datas):
             'blacklist_hostnames',
             ['dc.services.visualstudio.com'],
         )
-        response = requests.post(
-            url=self.options.endpoint,
-            data=json.dumps(envelopes),
-            headers={
-                'Accept': 'application/json',
-                'Content-Type': 'application/json; charset=utf-8',
-            },
-            timeout=self.options.timeout,
-        )
-        execution_context.set_opencensus_attr(
-            'blacklist_hostnames',
-            blacklist_hostnames,
-        )
-        response = response  # noqa
-        # print(response.status_code)
-        # print(response.text)
+        try:
+            response = requests.post(
+                url=self.options.endpoint,
+                data=json.dumps(envelopes),
+                headers={
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json; charset=utf-8',
+                },
+                timeout=self.options.timeout,
+            )
+        except Exception as ex:
```

Comment:
Added a TODO comment for now, will revisit when refactoring the exporter specific context (e.g. instead of having exporter manipulating the blacklist, having a dedicated context flag in exporter logic, so that integrations like requests would not intercept such activities and cause dead loop).